packages/backend/src/models/SpaceModel.ts

```diff
@@ -1672,29 +1708,45 @@ export class SpaceModel {
             projectUuid,
             userId,
             trx = this.database,
-            slug,
+            path,
         }: {
             trx?: Knex;
             userId: number;
             projectUuid: string;
-            slug?: string;
+            path?: string;
         },
     ): Promise<Space> {
         const [project] = await trx(ProjectTableName)
             .select('project_id')
             .where('project_uuid', projectUuid);
 
-        const spaceSlug =
-            slug ??
-            (await generateUniqueSpaceSlug(spaceData.name, project.project_id, {
-                trx,
-            }));
-        const spacePath = await this.generateSpacePath(
-            spaceSlug,
-            spaceData.parentSpaceUuid,
-            project.project_id,
-            { trx },
-        );
+        let spacePath = '';
+        let spaceSlug = '';
+
+        if (path) {
+            spacePath = path;
+            spaceSlug = await generateUniqueSpaceSlug(
+                spaceData.name,
+                project.project_id,
+                {
+                    trx,
+                },
+            );
+        } else {
+            spaceSlug = await generateUniqueSpaceSlug(
+                spaceData.name,
+                project.project_id,
+                {
+                    trx,
+                },
+            );
+            spacePath = await this.generateSpacePath(
+                spaceSlug,
+                spaceData.parentSpaceUuid,
+                project.project_id,
+                { trx },
+            );
+        }
```

Comment:
don't really understand this

Reply:
In `PromoteService` we always use `path` to match spaces across projects

1. `path` property is passed to `createSpace` function when Promoting a space or using content as code
2. If PromoteService or CoderService call `createSpace` that means that given space was not fount in given project so needs to be created – and existance check was done by using `path` (1)
3. So given all the above, it makes sense to create the new space using the passed `path` (instead of slug) as that's the field we use for matching

Reply:
the question then might be what's the role of the slug? In the case described above it seems redundant but we still have to generate it for consistency.